### ISSUE TITLE
Enhancement Request: Configurable CredentialOfferState in createVciIssuerBuilder

### ISSUE THREAD

**Issue by vijayshiyani** [Sphereon-Opensource/SSI-SDK#148]
I was examining the `createVciIssuerBuilder` function in the `functions.ts` file at [oid4vci-issuer package](https://github.com/Sphereon-Opensource/SSI-SDK/blob/develop/packages/oid4vci-issuer/src/functions.ts#L169) and noticed that it currently does not allow for a configurable `CredentialOfferState`. The function uses `withInMemoryCredentialOfferState()` which seems to set the `CredentialOfferState` to an in-memory state.

`builder.withInMemoryCredentialOfferState()`

However, in the [OID4VCI](https://github.com/Sphereon-Opensource/OID4VCI/tree/develop/packages/issuer) repository, I noticed that there is a capability to support a custom implementation of the state manager.

Is the current implementation in `createVciIssuerBuilder` deliberate or is there a plan to allow for a configurable `CredentialOfferState` in the future?

Allowing a configurable `CredentialOfferState` would provide more flexibility for different use cases.

Looking forward to your response.

**Comment by nklomp** [comment]
Yeah, that is something we will update. There is some refactor work on the low-level library coming up in which we will probably also simplify the state manager interface.

Before that we will have a look at whether it is simple to support custom `CredentialOfferState`. It mainly has to do with not really having other implementations yet and thus a bit of laziness ;)
